module/os/linux/zfs/zpl_xattr.c

```diff
@@ -1314,8 +1321,14 @@ __zpl_xattr_acl_set_access(struct inode *ip, const char *name,
 	if (ITOZSB(ip)->z_acl_type != ZFS_ACLTYPE_POSIX)
 		return (-EOPNOTSUPP);
 
+#if defined(HAVE_XATTR_SET_USERNS)
+	if (!zpl_inode_owner_or_capable(mnt_ns, ip))
+		return (-EPERM);
+#else
+	(void) mnt_ns;
 	if (!zpl_inode_owner_or_capable(kcred->user_ns, ip))
 		return (-EPERM);
+#endif
```

Comment:
You are right. zpl_inode_owner_or_capable() is an wrapper on inode_owner_or_capable().

I searched latest kernel code, there are a bunch of places where mnt_userns is passed into inode_owner_or_capable(). Are they all wrong? for example, this one: https://github.com/torvalds/linux/blob/master/fs/posix_acl.c#L1142

Without this change, doing "cp -p" of an idmapped file in a user namespace would fail.